03-debugging-w-chatgpt/debugging-w-chatgpt.md

```diff
@@ -0,0 +1,491 @@
+
+After completing the project, Althea has some time and wants to refactor their code. Just like with debugging help, if we ask for help refactoring, we want to use the same critical lens to check if ChatGPT’s response is truthful and solves what we are asking for. Althea came back to their chat above and asks:
+
+> **Althea**:  
+> Do you have suggestions for how I can simplify the code for choosing ten random elements from the list `pool`?
+
+> *ChatGPT*:  
+> Certainly! You can simplify the code for choosing ten random elements from the `pool` list using the `random.sample` function. The `random.sample` function is designed for this purpose and ensures that you get unique elements from the population.
+>
+> Here's the updated code:
+> 
+> ```python
+> import random
+> 
+> def draw_letters():
+>     distribution = {'A': 9, 'B': 2, 'O': 8, 'C': 2, 'P': 2, 
+>                     'D': 4, 'Q': 1, 'E': 12, 'R': 6, 'F': 2, 
+>                     'S': 4, 'G': 3, 'T': 6, 'H': 2, 'U': 4, 
+>                     'I': 9, 'V': 2, 'J': 1, 'W': 2, 'K': 1, 
+>                     'X': 1, 'L': 4, 'Y': 2, 'M': 2, 'Z': 1}
+>     
+>     pool = []
+>     for letter in distribution:
+>         pool.extend([letter] * distribution[letter])
+> 
+>     hand = random.sample(pool, 10)
+> 
+>     return hand
+> ```
+> 
+> This code uses `random.sample(pool, 10)` to directly obtain a list of 10 unique elements from the `pool` list, eliminating the need for the loop and manual removal of elements.
+
+We can see that ChatGPT is suggesting that Althea uses a separate function `sample` from the `random` module. Before copy/pasting the suggestion into their project, they go back to the random module documentation or other trusted resources to ensure: 
+- Does the `random.sample` function exist? 
+- If `random.sample` does exist, does it actually do what ChatGPT thinks it does? 
+- If it does what ChatGPT thinks, then does using the function meet the original intention of the code we’re trying to fix?
+
+In this case, the answer to all of these questions is Yes! Althea can once again update their code and run the test suite to ensure everything is still in working order after their changes 🎉.
+
+## Summary
+
+ChatGPT can be a really helpful tool to give us more insight on what code is doing, but we need to be careful about over-reliance. There are many situations in life such as interviews and code review meetings where we may not have access to generative AI tools, meaning that tools like ChatGPT are best used to support and augment the debugging skills we’ve built rather than to replace them. When we start out a debugging journey, we still want to begin with searching errors on the web, using print statements, and setting breakpoints to step through our code. 
```

Comment:
Could we add 'asking others for help' too?

"When we start out a debugging journey, we still want to begin with searching errors on the web, using print statements, setting breakpoints to step through our code, and asking others for help before turning to ChatGPT."